### ISSUE TITLE
JWPlayer is Updating UI on background thread

### ISSUE THREAD

**Issue by adamszeptycki** [jwplayer/jwplayer-sdk-ios-demo#64]
```
Main Thread Checker: UI API called on a background thread: -[UIView setHidden:]
PID: 21675, TID: 4345275, Thread name: (none), Queue name: JWSDKRouter, QoS: 0
Backtrace:
4   Luve                                0x000000010a1ef909 $s4Luve25VideoPlayerViewControllerC19onControlBarVisibleyySo15JWControlsEvent_So7JWEventCXcSgF + 441
5   Luve                                0x000000010a1ef96c $s4Luve25VideoPlayerViewControllerC19onControlBarVisibleyySo15JWControlsEvent_So7JWEventCXcSgFTo + 60
6   JWPlayer_iOS_SDK                    0x000000010ea6e6b3 -[JWAVPlayerHandler controlBarIsVisible:] + 582
7   JWPlayer_iOS_SDK                    0x000000010eab2aa4 -[JWSDKRouter controlBarIsVisible:] + 57
8   libdispatch.dylib                   0x0000000115967d7f _dispatch_call_block_and_release + 12
9   libdispatch.dylib                   0x0000000115968db5 _dispatch_client_callout + 8
10  libdispatch.dylib                   0x0000000115970225 _dispatch_lane_serial_drain + 778
11  libdispatch.dylib                   0x0000000115970e9c _dispatch_lane_invoke + 425
12  libdispatch.dylib                   0x000000011597aea3 _dispatch_workloop_worker_thread + 733
13  libsystem_pthread.dylib             0x0000000115d4c611 _pthread_wqthread + 421
14  libsystem_pthread.dylib             0x0000000115d4c3fd start_wqthread + 13
```
I keep getting this when ads start playing.

**Comment by adamszeptycki** [comment]
Or this one is happening when I navigate from video (I disabled ads):
```
Main Thread Checker: UI API called on a background thread: -[WKWebView removeFromSuperview]
PID: 66307, TID: 5754379, Thread name: (none), Queue name: com.apple.root.default-qos, QoS: 0
Backtrace:
4   JWPlayer_iOS_SDK                    0x0000000107ed6cf4 -[JWPlayerController dealloc] + 268
5   Foundation                          0x00000001072b7590 NSKVODeallocate + 158
6   libobjc.A.dylib                     0x000000010cec372c _ZN11objc_object17sidetable_releaseEb + 202
7   JWPlayer_iOS_SDK                    0x0000000107ebc8e2 __destroy_helper_block_e8_32s40s + 24
8   libsystem_blocks.dylib              0x000000010edcd9cd _Block_release + 105
9   libdispatch.dylib                   0x000000010ecd2db5 _dispatch_client_callout + 8
10  libdispatch.dylib                   0x000000010ecd57b9 _dispatch_queue_override_invoke + 1022
11  libdispatch.dylib                   0x000000010ece3632 _dispatch_root_queue_drain + 351
12  libdispatch.dylib                   0x000000010ece3fca _dispatch_worker_thread2 + 130
13  libsystem_pthread.dylib             0x000000010f0bb6b3 _pthread_wqthread + 583
14  libsystem_pthread.dylib             0x000000010f0bb3fd start_wqthread + 13

```

**Comment by karimMourra** [comment]
Hi @adamszeptycki ! thank you for reporting, we are trying to include a fix for 3.6.0. We will confirm if the fix is done in time for the release, otherwise it will ship with 3.6.1

**Comment by karimMourra** [comment]
Hi @adamszeptycki a fix should have shipped with version 3.6.0. Please reopen if your issue persists
